### PR TITLE
fix(fgs): fix fgs instance resource lint error 

### PR DESCRIPTION
**Pull request by MrLiangSir**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:
 ./scripts/codecheck.sh ./huaweicloud/services/fgs           
==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        6      2651      209        69     2373        289            64.49
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~s/fgs/resource_huaweicloud_fgs_function.go       942       74        34      834        123            14.75
~es/fgs/resource_huaweicloud_fgs_trigger.go       919       64        23      832        100            12.02
~weicloud_fgs_async_invoke_configuration.go       221       23         0      198         24            12.12
~fgs/resource_huaweicloud_fgs_dependency.go       164       19         2      143         18            12.59
~s/data_source_huaweicloud_fgs_functions.go       265       17         8      240         16             6.67
~ata_source_huaweicloud_fgs_dependencies.go       140       12         2      126          8             6.35
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     6      2651      209        69     2373        289            64.49
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 79006 bytes, 0.079 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
13 fgs resourceFgsFunctionV2Create huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:392:1
12 fgs resourceFgsFunctionV2Update huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:722:1
10 fgs resourceFgsFunctionV2MetadataUpdate huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:789:1
9 fgs buildFunctionGraphTriggerParameters huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go:566:1
8 fgs setTriggerEventData huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go:759:1
8 fgs resourceFunctionGraphTriggerCreate huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go:601:1
8 fgs buildFgsFunctionV2Parameters huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:335:1
7 fgs triggerV2StateRefreshFunc huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go:862:1
7 fgs resourceFunctionGraphTriggerRead huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go:792:1
7 fgs dataSourceFunctionGraphFunctionsRead huaweicloud/services/fgs/data_source_huaweicloud_fgs_functions.go:169:1
Average: 3.32

==> Checking for golangci-lint...

==> Checking for Nolint directives...
./huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:413:    // lintignore:R019
./huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:731:    // lintignore:R019
./huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:738:    // lintignore:R019

==> Checking for TF features in fgs...
  -> the resource in resource_huaweicloud_fgs_trigger.go should can be imported

==> Checking for misspell in fgs...
examples/fgs/obs-download/main.tf:46:64: "directoty" is a misspelling of "directory"

==> Checking for code complexity in ./huaweicloud/services/acceptance/fgs...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        6      1976      130         8     1838          8             4.43
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~oud_fgs_async_invoke_configuration_test.go       186       10         2      174          2             1.15
~esource_huaweicloud_fgs_dependency_test.go        94       10         0       84          2             2.38
~/resource_huaweicloud_fgs_function_test.go       872       49         3      820          2             0.24
~s/resource_huaweicloud_fgs_trigger_test.go       331       24         2      305          2             0.66
~ource_huaweicloud_fgs_dependencies_test.go        86       12         0       74          0             0.00
~a_source_huaweicloud_fgs_functions_test.go       407       25         1      381          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     6      1976      130         8     1838          8             4.43
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 58783 bytes, 0.059 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
2 fgs getTriggerResourceFunc huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go:16:1
2 fgs getResourceObj huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go:17:1
2 fgs getDependencyResourceFunc huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_dependency_test.go:16:1
2 fgs getAsyncInvokeConfigFunc huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_async_invoke_configuration_test.go:17:1
1 fgs testAccFunctionGraphLtsTrigger_basic huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go:287:1
Average: 1.06

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/fgs...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/fgs...
./huaweicloud/services/acceptance/fgs/data_source_huaweicloud_fgs_functions_test.go:50: //nolint:revive
./huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go:287:    //nolint:revive
./huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go:309:    //nolint:revive
./huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go:365:    //nolint:revive
./huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go:185:     //nolint:revive
./huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go:288:     //nolint:revive

==> Cleanup patch...

Check Completed!

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

 make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run TestAccFgsV2Function_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run TestAccFgsV2Function_basic -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_basic
=== PAUSE TestAccFgsV2Function_basic
=== CONT  TestAccFgsV2Function_basic
--- PASS: TestAccFgsV2Function_basic (88.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       88.848s
 make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run TestAccFunctionGraphTrigger_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run TestAccFunctionGraphTrigger_basic -timeout 360m -parallel 4
=== RUN   TestAccFunctionGraphTrigger_basic
=== PAUSE TestAccFunctionGraphTrigger_basic
=== CONT  TestAccFunctionGraphTrigger_basic
--- PASS: TestAccFunctionGraphTrigger_basic (37.46s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       37.504s
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run TestAccAsyncInvokeConfig_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run TestAccAsyncInvokeConfig_basic -timeout 360m -parallel 4
=== RUN   TestAccAsyncInvokeConfig_basic
=== PAUSE TestAccAsyncInvokeConfig_basic
=== CONT  TestAccAsyncInvokeConfig_basic
    acceptance.go:389: HW_FGS_TRIGGER_LTS_AGENCY must be set for FGS trigger acceptance tests
--- SKIP: TestAccAsyncInvokeConfig_basic (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       0.045s
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run TestAccFunctionGraphResourceDependency_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run TestAccFunctionGraphResourceDependency_basic -timeout 360m -parallel 4
=== RUN   TestAccFunctionGraphResourceDependency_basic
=== PAUSE TestAccFunctionGraphResourceDependency_basic
=== CONT  TestAccFunctionGraphResourceDependency_basic
    acceptance.go:492: HW_OBS_BUCKET_NAME must be set for OBS object acceptance tests
--- SKIP: TestAccFunctionGraphResourceDependency_basic (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       0.043s

### CODE REVIEW

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go [hunk 1]
```diff
@@ -14,6 +14,12 @@ import (
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/common"
 )
 
+const (
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go]: it's better to use `nolint` to ignore the lint error

File: huaweicloud/services/fgs/data_source_huaweicloud_fgs_dependencies.go [hunk 2]
```diff
@@ -82,12 +85,12 @@ func DataSourceFunctionGraphDependencies() *schema.Resource {
...
-	config := meta.(*config.Config)
-	region := config.GetRegion(d)
-	client, err := config.FgsV2Client(region)
+func dataSourceFunctionGraphDependenciesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.FgsV2Client(region)
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud compute client: %s", err)
+		return diag.Errorf("error creating compute client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/fgs/data_source_huaweicloud_fgs_dependencies.go]: **compute client** is not correct

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 3]
```diff
@@ -31,7 +29,7 @@ func ResourceFgsFunctionV2() *schema.Resource {
 		Update: resourceFgsFunctionV2Update,
```
Comment (ShiChangkuo) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: please use the function with Context

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go [hunk 4]
```diff
@@ -1,29 +1,30 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 
 	"github.com/chnsz/golangsdk"
-	dedicatedGroups "github.com/chnsz/golangsdk/openstack/apigw/dedicated/v2/apigroups"
-	dedicatedEnvs "github.com/chnsz/golangsdk/openstack/apigw/dedicated/v2/environments"
+	grou "github.com/chnsz/golangsdk/openstack/apigw/dedicated/v2/apigroups"
+	envs "github.com/chnsz/golangsdk/openstack/apigw/dedicated/v2/environments"
```
Comment (ShiChangkuo) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go]: why do we change the alias of packages?

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go [hunk 5]
```diff
@@ -42,7 +42,7 @@ func TestAccFgsV2Function_basic(t *testing.T) {
 				Config: testAccFgsV2Function_basic(randName),
 				Check: resource.ComposeTestCheckFunc(
 					rc.CheckResourceExists(),
-					resource.TestCheckResourceAttr(resourceName, "functiongraph_version", "v1"),
+					resource.TestCheckResourceAttr(resourceName, "functiongraph_version", "v2"),
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go]: it seems that the default value was changed, so we should update the docs.

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go [hunk 6]
```diff
@@ -307,15 +308,15 @@ resource "huaweicloud_identity_agency" "test" {
...
+  name        = "%s"
   app         = "default"
   handler     = "index.handler"
   memory_size = 128
   timeout     = 10
   runtime     = "Python2.7"
   code_type   = "inline"
   agency      = huaweicloud_identity_agency.test.name
-  func_code   = "aW1wb3J0IGpzb24KZGVmIGhhbmRsZXIgKGZW50LCBjb250ZXh0KToKICAgIG91dHB1dCA9ICdIZWxsbyBtZXNzYWdlOiAnICsganNvbi5kdW1wcyhldmVudCkKICAgIHJldHVybiBvdXRwdXQ="
+  func_code   = "aW1wb3J0IGpzb24KZGVmIGhhbmRsZXIgKGV2ZW50LCBjb250ZXh0KToKICAgIG91dHB1dCA9ICdIZWxsbyBtZXNzYWdlOiAnICsganNvbi5kdW1wcyhldmVudCkKICAgIHJldHVybiBvdXRwdXQ="
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go]: why we change the value of func_code?

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 7]
```diff
@@ -389,38 +389,38 @@ func buildFgsFunctionV2Parameters(d *schema.ResourceData, config *config.Config)
...
 	// in terraform, we convert to id, not using FuncUrn
 	d.SetId(f.FuncUrn)
 	urn := resourceFgsFunctionUrn(d.Id())
-	//lintignore:R019
+	// lintignore:R019
 	if d.HasChanges("vpc_id", "func_mounts", "app_agency", "initializer_handler", "initializer_timeout") {
 		err := resourceFgsFunctionV2MetadataUpdate(fgsClient, urn, d)
 		if err != nil {
-			return err
+			return diag.Errorf("error resource fgs function v2 metadata update: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: the error message is hard to understand, please refer others

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go (hunk 6 above)
Comment (MrLiangSir) [reply huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go]: problem solved

File: huaweicloud/services/acceptance/fgs/data_source_huaweicloud_fgs_dependencies_test.go [hunk 8]
```diff
@@ -66,25 +65,22 @@ func TestAccFunctionGraphDependencies_runtime(t *testing.T) {
...
-	return fmt.Sprintf(`
-data "huaweicloud_fgs_dependencies" "test" {
+	result := `data "huaweicloud_fgs_dependencies" "test" {
   type = "public"
   name = "obssdk-3.0.2"
-}
-`)
+  }`
+	return result
 }
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/fgs/data_source_huaweicloud_fgs_dependencies_test.go]:
```suggestion
const testAccFunctionGraphDependencies_name = `
data "huaweicloud_fgs_dependencies" "test" {
  type = "public"
  name = "obssdk-3.0.2"
}`
```

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 9]
```diff
@@ -645,7 +645,7 @@ func resourceFgsFunctionV2Read(d *schema.ResourceData, meta interface{}) error {
 		d.Set("versions", versionConfig),
 	)
 	if err := mErr.ErrorOrNil(); err != nil {
-		return fmtp.Errorf("Error setting vault fields: %s", err)
+		return diag.Errorf("error setting vault fields: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]:
```suggestion
		return diag.Errorf("error setting function fields: %s", err)
```

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go (hunk 5 above)
Comment (Lance52259) [inline huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go]: the default value is `v1` in `cn-north-4`, please revert it and append a comment ahead of this check step.

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go [hunk 10]
```diff
@@ -42,7 +43,10 @@ func TestAccFgsV2Function_basic(t *testing.T) {
 				Config: testAccFgsV2Function_basic(randName),
 				Check: resource.ComposeTestCheckFunc(
 					rc.CheckResourceExists(),
-					resource.TestCheckResourceAttr(resourceName, "functiongraph_version", "v1"),
+					// Default value is v2. Some regions support only v1, the default value is v1
+					resource.ComposeTestCheckFunc(
+						resource.TestMatchResourceAttr(resourceName, "functiongraph_version", regexp.MustCompile(`v1|v2`)),
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go]: it seems that we can use `TestMatchResourceAttr` directly

**Pull request merged**